src/middleware.ts

```diff
@@ -0,0 +1,162 @@
+import {NextRequest, NextResponse} from 'next/server'
+import {ACCESS_TOKEN_KEY, EGGHEAD_USER_COOKIE_KEY} from './config'
+import {loadCio} from './lib/customer'
+import {loadUser} from './lib/users'
+
+const PUBLIC_FILE = /\.(.*)$/
+
+const CIO_COOKIE_KEY = 'cio_id'
+
+// Supports both a single string value or an array of matchers
+export const config = {
+  matcher: ['/', '/pricing'],
+}
+
+/**
+ * with this approach, logged in users can be shown
+ * '/' and anon users '/signup' IN PLACE
+ *
+ * we don't have a server cookie to get member status, but that
+ * should be on the short list. That would make this even better
+ * because we could send them into a more specific /learn route
+ * and we could start personalizing that a bit more using various
+ * cookies and such.
+ *
+ * This looks a lot like the i18n example:
+ * https://github.com/vercel/examples/tree/main/edge-functions/i18n
+ *
+ * an the split testing example that puts them in a bucket:
+ * https://github.com/vercel/examples/edge-functions/ab-testing-simple
+ *
+ */
+export async function middleware(req: NextRequest) {
```

Comment:
this file was the biggest change since middleware isn't nested anymore with next 12.2 we needed to consolidate the two routes we use middleware for

https://nextjs.org/docs/messages/middleware-upgrade-guide#no-nested-middleware

Reply:
This looks good. if it ends up with more route blocks, we could explore breaking the blocks out into separate functions/modules that get invoked here. That'd make it easier to glance at what is happening and also encapsulate the specifics of each (e.g. inputs and output).